From 5ccbe6d8b91ad943ce257a2338d93724d9dc33c0 Mon Sep 17 00:00:00 2001
From: Ondrej Spacek <ondrej.spacek@nxp.com>
Date: Thu, 11 Nov 2021 14:04:22 +0100
Subject: [PATCH 22/24] dts: s32g: Allow simpler interrupt pass through for
 hypervisor

commit 5ccbe6d8b91ad943ce257a2338d93724d9dc33c0 from
https://source.codeaurora.org/external/autobsps32/linux

PFE device tree is currently to complicated to be correctly handled by
Xen. Xen can't parse interrupts from nodes that do not have valid
registers to be passed through together with the interrupt.

As a temporary workaround copy interrupts property to root node of
the slave.

Isuse: ALB-7539
Upstream-Status: Pending 

Signed-off-by: Ondrej Spacek <ondrej.spacek@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-s32g.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi
index 514eb5f6adbd..c5395ed9bf03 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi
@@ -312,6 +312,8 @@
 		#address-cells = <0x01>;
 		#size-cells = <0x00>;
 		interrupt-parent = <&gic>;
+		/* Interrupt duplicated for XEN passthrough */
+		interrupts = <GIC_SPI 193 IRQ_TYPE_EDGE_RISING>;
 		dma-coherent;
 		memory-region = <&pfe_reserved_slave>,
 				<&pfe_reserved_bdr_slave>;
-- 
2.17.1

